generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -0,0 +1,274 @@
+#include "generator/srtm_parser.hpp"
+
+#include "routing/bicycle_model.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "geometry/mercator.hpp"
+#include "geometry/point2d.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "indexer/feature.hpp"
+#include "indexer/feature_processor.hpp"
+#include "indexer/map_style_reader.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+
+#include "3party/gflags/src/gflags/gflags.h"
+#include "3party/kdtree++/kdtree.hpp"
+
+#include "std/iostream.hpp"
+#include "std/fstream.hpp"
+#include "std/map.hpp"
+#include "std/string.hpp"
+
+DEFINE_string(srtm_path, "", "Path to directory with SRTM files");
+DEFINE_string(mwm_path, "", "Path to an mwm file.");
+
+namespace
+{
+routing::BicycleModel const & GetBicycleModel()
+{
+  static routing::BicycleModel const instance;
+  return instance;
+}
+
+int32_t Coord2RoughCoord(double d)
+{
+  int32_t constexpr kFactor = 100000;
+  return static_cast<int32_t>(round(d * kFactor));
+}
+
+struct RoughPoint
+{
+  RoughPoint(m2::PointD const & point)
+    : x(Coord2RoughCoord(point.x)), y(Coord2RoughCoord(point.y)) {}
+
+  int32_t x;
+  int32_t y;
+};
+
+bool operator< (RoughPoint const & l, RoughPoint const & r)
+{
+  if (l.x != r.x)
+    return l.x < r.x;
+  return l.y < r.y;
+}
+
+template <typename Cont>
+void PrintCont(Cont const & cont, string const & title, string const & msgText1,
+               string const & msgText2)
+{
+  cout << endl << title << endl;
+  for (auto const & a : cont)
+    cout << a.second << msgText1 << a.first << msgText2 << endl;
+}
+
+template <typename Cont>
+void WriteCSV(Cont const & cont, string const & fileName)
+{
+  ofstream fout(fileName);
+  for (auto const & a : cont)
+    fout << a.first << ";" << a.second << endl;
```

Comment:
According to RTF 4180 (https://www.ietf.org/rfc/rfc4180.txt) - yes. But semi-comma is a default separator in Open Office 4 and these csvs are used to make charts with Open Office. On the other hand less restricted csv format is often used: https://en.wikipedia.org/wiki/Comma-separated_values